torch/csrc/jit/script/compiler.cpp

```diff
@@ -34,23 +34,127 @@ using ListAttributeMap = std::unordered_map<
     std::string,
     std::pair<const std::vector<double>, std::string>>;
 
+// Keep track of environment as we descend down nested control
+// structures.
+struct Environment {
```

Comment:
Here, one must be careful to distinguish the concept of environment/scope in the frontend language (NOT this data structure) and its concept in the desugared language (THIS data structure, sort of). This is especially is poignant here since the Environment data structure is used to intermediate between the desugaring translation.

Perhaps a good start is to reword the comment. Maybe something like:

```
// Auxiliary data structure for desugaring variable binding into our always
// explicitly scoped language as we descend down
// nested control structures in the frontend (which themselves don't introduce
// scopes)
```

Reply:
It may also be helpful to future readers to mention that this is for conversion to SSA form.

Reply:
Also, it would be nice to document the  whole `captured_inputs` + `deleteExtraInputs` business